v2/dtos/device.go

```diff
@@ -34,15 +34,15 @@ type Device struct {
 // UpdateDevice and its properties are defined in the APIv2 specification:
 // https://app.swaggerhub.com/apis-docs/EdgeXFoundry1/core-metadata/2.x#/UpdateDevice
 type UpdateDevice struct {
-	Id             *string                       `json:"id" validate:"required_without=Name"`
-	Name           *string                       `json:"name" validate:"required_without=Id"`
-	Description    *string                       `json:"description"`
+	Id             *string                       `json:"id" validate:"required_without=Name,edgex-update-dto-id"`
+	Name           *string                       `json:"name" validate:"required_without=Id,edgex-update-dto-name"`
+	Description    *string                       `json:"description" validate:"omitempty,gt=0"`
 	AdminState     *string                       `json:"adminState" validate:"omitempty,oneof='LOCKED' 'UNLOCKED'"`
 	OperatingState *string                       `json:"operatingState" validate:"omitempty,oneof='ENABLED' 'DISABLED'"`
 	LastConnected  *int64                        `json:"lastConnected"`
 	LastReported   *int64                        `json:"lastReported"`
-	ServiceName    *string                       `json:"serviceName"`
-	ProfileName    *string                       `json:"profileName"`
+	ServiceName    *string                       `json:"serviceName" validate:"omitempty,gt=0"`
+	ProfileName    *string                       `json:"profileName" validate:"omitempty,gt=0"`
```

Comment:
Why not use `edgex-update-dto-name` instead of `gt=0` so string of blanks is caught.

Reply:
Modified. You're right, we should check the blank string.